state/pgstatestorage/pgstatestorage.go

```diff
@@ -0,0 +1,370 @@
+	const insertTrustedReorgSQL = "INSERT INTO state.trusted_reorg (timestamp, batch_num, reason) VALUES (NOW(), $1, $2)"
+
+	e := p.getExecQuerier(dbTx)
+	_, err := e.Exec(ctx, insertTrustedReorgSQL, reorg.BatchNumber, reorg.Reason)
+	return err
+}
+
+// CountReorgs returns the number of reorgs
+func (p *PostgresStorage) CountReorgs(ctx context.Context, dbTx pgx.Tx) (uint64, error) {
+	const countReorgsSQL = "SELECT COUNT(*) FROM state.trusted_reorg"
+
+	var count uint64
+	q := p.getExecQuerier(dbTx)
+	err := q.QueryRow(ctx, countReorgsSQL).Scan(&count)
+	if err != nil {
+		return 0, err
+	}
+	return count, nil
+}
+
+// GetReorgedTransactions returns the transactions that were reorged
+func (p *PostgresStorage) GetReorgedTransactions(ctx context.Context, batchNumber uint64, dbTx pgx.Tx) ([]*types.Transaction, error) {
+	const getReorgedTransactionsSql = "SELECT encoded FROM state.transaction t INNER JOIN state.l2block b ON t.l2_block_num = b.block_num WHERE b.batch_num >= $1 ORDER BY l2_block_num ASC"
+	e := p.getExecQuerier(dbTx)
+	rows, err := e.Query(ctx, getReorgedTransactionsSql, batchNumber)
+	if !errors.Is(err, pgx.ErrNoRows) && err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	txs := make([]*types.Transaction, 0, len(rows.RawValues()))
+
+	for rows.Next() {
+		if rows.Err() != nil {
+			return nil, rows.Err()
+		}
+		var encodedTx string
+		err := rows.Scan(&encodedTx)
+		if err != nil {
+			return nil, err
+		}
+
+		tx, err := state.DecodeTx(encodedTx)
+		if err != nil {
+			return nil, err
+		}
+		txs = append(txs, tx)
+	}
+	return txs, nil
+}
+
+// GetNativeBlockHashesInRange return the state root for the blocks in range
+func (p *PostgresStorage) GetNativeBlockHashesInRange(ctx context.Context, fromBlock, toBlock uint64, dbTx pgx.Tx) ([]common.Hash, error) {
+	const l2TxSQL = `
+    SELECT l2b.state_root
+      FROM state.l2block l2b
+     WHERE block_num BETWEEN $1 AND $2
+     ORDER BY l2b.block_num ASC`
+
+	if toBlock < fromBlock {
+		return nil, state.ErrInvalidBlockRange
+	}
+
+	blockRange := toBlock - fromBlock
+	if p.cfg.MaxNativeBlockHashBlockRange > 0 && blockRange > p.cfg.MaxNativeBlockHashBlockRange {
+		return nil, state.ErrMaxNativeBlockHashBlockRangeLimitExceeded
+	}
+
+	e := p.getExecQuerier(dbTx)
+	rows, err := e.Query(ctx, l2TxSQL, fromBlock, toBlock)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	nativeBlockHashes := []common.Hash{}
+
+	for rows.Next() {
+		var nativeBlockHash string
+		err := rows.Scan(&nativeBlockHash)
+		if err != nil {
+			return nil, err
+		}
+		nativeBlockHashes = append(nativeBlockHashes, common.HexToHash(nativeBlockHash))
+	}
+	return nativeBlockHashes, nil
+}
+
+// GetBatchL2DataByNumber returns the batch L2 data of the given batch number.
+func (p *PostgresStorage) GetBatchL2DataByNumber(ctx context.Context, batchNumber uint64, dbTx pgx.Tx) ([]byte, error) {
+	getBatchL2DataByBatchNumber := "SELECT raw_txs_data FROM state.batch WHERE batch_num = $1"
```

Comment:
Does it make sense to extract this into constant?

Reply:
modified to be a constant. But wee decided to put the queries inside each function. Before we used to have a massive `const` statement with all the sql, and it was painful for the queries and the functions that called the queries to be so separated in space. There is almost an exact 1:1 relation between query and function